lib/msal-core/src/telemetry/HttpEvent.ts

```diff
@@ -0,0 +1,73 @@
+import TelemetryEvent from "./TelemetryEvent";
+import { EVENT_NAME_PREFIX } from "./TelemetryConstants";
+import { QPDict } from "../AuthenticationParameters";
+import { Utils } from "../Utils";
+import { scrubTenantFromUri } from "./TelemetryUtils";
+
+export const HTTP_PATH_KEY = `${EVENT_NAME_PREFIX}http_path`;
+export const USER_AGENT_KEY = `${EVENT_NAME_PREFIX}user_agent`;
+export const QUERY_PARAMETERS_KEY = `${EVENT_NAME_PREFIX}query_parameters`;
+export const API_VERSION_KEY = `${EVENT_NAME_PREFIX}api_version`;
+export const RESPONSE_CODE_KEY = `${EVENT_NAME_PREFIX}response_code`;
+export const O_AUTH_ERROR_CODE_KEY = `${EVENT_NAME_PREFIX}oauth_error_code`;
+export const HTTP_METHOD_KEY = `${EVENT_NAME_PREFIX}http_method`;
+export const REQUEST_ID_HEADER_KEY = `${EVENT_NAME_PREFIX}request_id_header`;
+export const SPE_INFO_KEY = `${EVENT_NAME_PREFIX}spe_info`;
+export const SERVER_ERROR_CODE_KEY = `${EVENT_NAME_PREFIX}server_error_code`;
+export const SERVER_SUB_ERROR_CODE_KEY = `${EVENT_NAME_PREFIX}server_sub_error_code`;
+
+export default class HttpEvent extends TelemetryEvent {
+
+    constructor(correlationId: string) {
+        super(`${EVENT_NAME_PREFIX}http_event`, correlationId);
+    }
+
+
+    // Believe this should be whole url, not just a path
+    public set httpPath(httpPath: string) {
+        this.event[HTTP_PATH_KEY] = scrubTenantFromUri(httpPath).toLowerCase();
+    }
+
+    public set userAgent(userAgent: string) {
+        this.event[USER_AGENT_KEY] = userAgent;
+    }
+
+    public set queryParams(queryParams: QPDict) {
+        this.event[QUERY_PARAMETERS_KEY] = Utils.generateQueryParametersString(queryParams);
+    }
+
+    public set apiVersion(apiVersion: string) {
+        this.event[API_VERSION_KEY] = apiVersion.toLowerCase();
+    }
+
+    public set httpResponseStatus(statusCode: number) {
+        this.event[RESPONSE_CODE_KEY] = statusCode;
```

Comment:
undecided to benefit of setters / chaining, willing to be convinced

Reply:
While I don't feel strongly about this one way or the other, and there are some benefits to chaining,  I think it is better to follow JS Best practices here. It may mess up inheritance and in general doesn't follow command-query separation principle, so I would suggest using setters.

Reply:
As an internal implementation, I am fine shipping this, and if we want to update implementation to be cleaner I am fine changing down the road